### ISSUE TITLE
Make fonts const in lvgl format

### ISSUE THREAD

**Issue by kisvegabor** [lvgl/lv_font_conv#59]
As discussed [here](https://forum.lvgl.io/t/make-font-cache-optional/4643) the lvgl fonts could be `const` with a few modifications.

@puzrin If you have some time could you modify the end of the generated fonts like this?
See the parts added with `#if LV_VERSION_CHECK(8, 0, 0)`.

```c
/*--------------------
 *  ALL CUSTOM DATA
 *--------------------*/
#if LV_VERSION_CHECK(8, 0, 0)
/*Store all the custom data of the font*/
static  lv_font_fmt_txt_glyph_cache_t cache;
static const lv_font_fmt_txt_dsc_t font_dsc = {
#else
static lv_font_fmt_txt_dsc_t font_dsc = {
#endif
    .glyph_bitmap = gylph_bitmap,
    .glyph_dsc = glyph_dsc,
    .cmaps = cmaps,
    .kern_dsc = &kern_classes,
    .kern_scale = 16,
    .cmap_num = 2,
    .bpp = 4,
    .kern_classes = 1,
    .bitmap_format = 0,
#if LV_VERSION_CHECK(8, 0, 0)
    .cache = &cache
#endif
};

/*-----------------
 *  PUBLIC FONT
 *----------------*/

/*Initialize a public general font descriptor*/
#if LV_VERSION_CHECK(8, 0, 0)
const lv_font_t lv_font_montserrat_14 = {
#else
lv_font_t lv_font_montserrat_14 = {
#endif
    .get_glyph_dsc = lv_font_get_glyph_dsc_fmt_txt,    /*Function pointer to get glyph's data*/
    .get_glyph_bitmap = lv_font_get_bitmap_fmt_txt,    /*Function pointer to get glyph's bitmap*/
    .line_height = 16,          /*The maximum line height required by the font*/
    .base_line = 3,             /*Baseline measured from the bottom of the line*/
#if !(LVGL_VERSION_MAJOR == 6 && LVGL_VERSION_MINOR == 0)
    .subpx = LV_FONT_SUBPX_NONE,
#endif
#if LV_VERSION_CHECK(7, 4, 0)
    .underline_position = -1,
    .underline_thickness = 1,
#endif
    .dsc = &font_dsc           /*The custom font data. Will be accessed by `get_glyph_bitmap/dsc` */
};
```

**Comment by puzrin** [comment]
No problem. A couple of questions before start:

- Does this conditional magic produce enough of added value? Memory savings do not seems impressive.
- There are chances switch to binary format. If those are high, i would avoid extra modifications with short life cycle.

Just confirm your request, it's not difficult for me.

**Comment by kisvegabor** [comment]
> Does this conditional magic produce enough of added value? Memory savings do not seems impressive.

Only 52 bytes RAM saving on every font. With 6 fonts, 312 bytes. It's not a game-changer,  but if we can save some RAM for free, why not?

> There are chances switch to binary format. If those are high, i would avoid extra modifications with short life cycle.

I still "believe" in the text format. :slightly_smiling_face: 
Keeping all font data `const` is quite reasonable so I don't think these changes will have a short life cycle.

**Comment by puzrin** [comment]
https://github.com/lvgl/lv_font_conv/commit/77b69e62b70887758d25000e79ce425a2d4f04b9

Try dev branch. If it's ok, i will merge & publish.

**Comment by kisvegabor** [comment]
Thanks!
See https://github.com/lvgl/lv_font_conv/commit/77b69e62b70887758d25000e79ce425a2d4f04b9#r47995245

**Comment by puzrin** [comment]
https://github.com/lvgl/lv_font_conv/commit/3b168802e2527e797f70f6a2814d7960eeea6710

Force-pushed fix to dev. Anything else?

**Comment by kisvegabor** [comment]
Works well, thank you!

**Comment by puzrin** [comment]
Released 1.5.0
